File: learning-api/src/courses/getCourse.mjs
import { DynamoDBClient } from "@aws-sdk/client-dynamodb";
import { DynamoDBDocumentClient, GetCommand } from "@aws-sdk/lib-dynamodb";

const client = new DynamoDBClient();
const docClient = DynamoDBDocumentClient.from(client);

export const handler = async (event) => {
    // get course id from path parameters
    const courseId = event.pathParameters.courseId;
    
    // set command parameters
    let params = {
        TableName: process.env.COURSES_TABLE,
        Key: { courseId }
    };
    
    // get data from DynamoDB
    const command = new GetCommand(params);
    const response = await docClient.send(command);
    console.log(response);

    if (response.Item) {
        return {
            statusCode: 200,
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify(response.Item)
        };
    }
    else {
        return {
            statusCode: 404,
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({ message: 'Course not found' })
        };
    }
};

File: learning-api-python/src/courses/get_course.py
import boto3
import os
import json

def handler(event, context):
    # get DynamoDB table
    dynamodb = boto3.resource("dynamodb")
    table = dynamodb.Table(os.getenv("COURSES_TABLE"))

    # get course id from path parameters
    courseId = event.get("pathParameters", {}).get("courseId")
    
    try:
        # get item from DynamoDB
        db_response = table.get_item(Key={"courseId": courseId})
        item = db_response.get("Item")
        # convert Decimal to float
        item["rating"] = float(item["rating"])
        response = {
            "statusCode": 200,
            "headers": { "Content-Type": "application/json" },
            "body": json.dumps(item)
        }
    except Exception as e:
        print(f'Error: {e}') # log the error
        response = {
            "statusCode": 500,
            "headers": { "Content-Type": "application/json" },
            "body": json.dumps({"message": "Failed to get course"})
        }
    return response